### PR TITLE
feat(AWS Kafka): Support `maximumBatchingWindow`

### PR DESCRIPTION
**Pull request by rm-hull**
As per #10580, this PR adds `maximumBatchingWindow` property to serverless.yaml for kafka event sources

### CODE REVIEW

File: docs/providers/aws/events/kafka.md [hunk 1]
```diff
@@ -141,3 +141,28 @@ functions:
...
+
+## Setting the BatchSize, MaximumBatchingWindow and StartingPosition
+
+You can set the `batchSize`, which effects how many messages can be processed in a single Lambda invocation. The default `batchSize` is 100, and the max `batchSize` is 10000.
+Likewise `maximumBatchingWindow` can be set to determine the amount of time the Lambda spends gathering records before invoking the function. The default is 0, but **if you set `batchSize` to more than 10, you must set `maximumBatchingWindow` to at least 1**. The maximum is 300.
+In addition, you can also configure `startingPosition`, which controls the position at which Lambda should start consuming messages from the topic. It supports two possible values, `TRIM_HORIZON` and `LATEST`, with `TRIM_HORIZON` being the default.
+
+In the following example, we specify that the `compute` function should have a `kafka` event configured with `batchSize` of 1000, `maximumBatchingWindow` of 30 seconds and `startingPosition` equal to `LATEST`.
+
+```yml
```
Comment (pgrzesik) [inline docs/providers/aws/events/kafka.md:153]: let's also make sure it's referenced in `serverless.yml.md` docs file

Comment (rm-hull) [reply docs/providers/aws/events/kafka.md:153]: There wasn't a section for kafka ... added one in 4deab48768fdce17ee1a54a93d3667c165be7f13

Comment (pgrzesik) [reply docs/providers/aws/events/kafka.md:153]: thanks 🙇

File: docs/providers/aws/guide/serverless.yml.md [hunk 2]
```diff
@@ -928,6 +928,35 @@ functions:
...
+          # See main kafka documentation for various access configuration settings
+          accessConfigurations:
+            # ...
+          # An array of bootstrap server addresses
+          bootstrapServers:
+            - abc3.xyz.com:9092
+            - abc2.xyz.com:9092
+          # name of Kafka topic to consume from
+          topic: MySelfManagedKafkaTopic
+          # Optional, must be in 1-10000 range
```
Comment (pgrzesik) [inline docs/providers/aws/guide/serverless.yml.md:950]: Let's be consistent in how we specify max and min values

**Review by pgrzesik — CHANGES_REQUESTED** [review]
Thank you @rm-hull - I have only small suggestion and it will be good to go

File: docs/providers/aws/guide/serverless.yml.md (hunk 2 above)
Comment (rm-hull) [reply docs/providers/aws/guide/serverless.yml.md:950]: Fixed in 814789c99055e66cb77606a1be5656e89271a343

**Review by pgrzesik — APPROVED** [review]
Thanks @rm-hull 🙇

**Pull request merged**